### PR TITLE
⚡️Spike new Icon story

### LINKED ISSUE #1240: Conform and document Storybook "components"
We have a lot of recurring styling/decorating (like wrappers and containers) components in our stories that need to be conformed and documented better.

To clarify that these are only Storybook components and not EDS components. When users now see them in the example codes, it's not clear that these are NOT EDS.

Fix: For example, we can use names like "Storybook_Wrapper" OR we can use decorators on each individual example. (Decorators do not show in code examples) Please note such decorators do not work as expected on the "control" examples (#1238)

### LINKED ISSUE #1850: ⚡️ Spike Storybook component story
With the knowledge and know how we've gathered so far, spike on how we want our component stories in Storybook to be.

Notes:
Suggested structure:
  1. Introduction (using description storybooks meta export, `parameters.docs.component.description`)
  2. Props table
  3. Installation
  4. Usage
  5. Accessibility  (if needed)
  6. Compact
  7. Rest: Other stories which might be relevant for specific properties/features of a component

### PR DESCRIPTION
**Pull request by mimarz**
resolves #1850 
resolves #1240

### CODE REVIEW

File: packages/eds-core-react/src/components/Icon/Icon.docs.mdx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+```tsx
+import { Icon } from '@equinor/eds-core-react'
+import { save } from '@equinor/eds-icons' // import "save" icon
+
+<Icon data={save} />
+```
+
+
+## Accessibility
+If you are using the icons semantically you need to provide the `title` property so that your icons are correctly accessible.
```
Comment (vnys) [inline packages/eds-core-react/src/components/Icon/Icon.docs.mdx:47]: “Using the icons semantically” should be rephrased, I see we have a similar description in the component itself that should also be updated. The use case is that if you’re using icons alone to convey meaning, you need to provide a title for screen reader users that reflect the action the icon represents. But if you already have icons and text, additional text is superfluous and you should rather use `aria-hidden="true"`.

**Review by vnys — CHANGES_REQUESTED** [review]
Good job 👍🏻 Just a couple of minor issues. And I think we should also add `title: 'Save', size: 24` to the story args, since we want to encourage devs to make their apps accessible, and in case the default values of radio button groups start working again at some point.

File: packages/eds-core-react/src/components/Icon/Icon.stories.tsx [hunk 2]
```diff
@@ -1,26 +1,29 @@
...
-  name: 'moon',
-  prefix: 'starwars',
-  height: '24',
-  width: '24',
-  svgPathData:
-    'M19.629,9.655c-0.021-0.589-0.088-1.165-0.21-1.723h-3.907V7.244h1.378V6.555h-2.756V5.866h2.067V5.177h-0.689V4.488h-1.378V3.799h0.689V3.11h-1.378V2.421h0.689V1.731V1.294C12.88,0.697,11.482,0.353,10,0.353c-5.212,0-9.446,4.135-9.629,9.302H19.629z M6.555,2.421c1.522,0,2.756,1.234,2.756,2.756S8.077,7.933,6.555,7.933S3.799,6.699,3.799,5.177S5.033,2.421,6.555,2.421z M12.067,18.958h-0.689v-0.689h2.067v-0.689h0.689V16.89h2.067v-0.689h0.689v-0.689h-1.378v-0.689h-2.067v-0.689h1.378v-0.689h2.756v-0.689h-1.378v-0.689h3.218c0.122-0.557,0.189-1.134,0.21-1.723H0.371c0.183,5.167,4.418,9.302,9.629,9.302c0.711,0,1.401-0.082,2.067-0.227V18.958z',
-}
+const Preview = styled.div`
+  box-shadow: rgb(0 0 0 / 10%) 0 1px 3px 0;
+  border: 1px solid rgba(0, 0, 0, 0.1);
```
[the remark this answers is not in the document]
Comment (mimarz) [reply packages/eds-core-react/src/components/Icon/Icon.stories.tsx]: I removed this as its no longer needed with the new storybook component structure.

File: packages/eds-core-react/package.json [hunk 3]
```diff
@@ -33,7 +33,7 @@
     "test": "tsc -p tsconfig.test.json && jest",
     "test:watch": "tsc-watch -p tsconfig.test.json --onFirstSuccess  \"jest --watch\"",
     "test:update-snapshots": "jest --updateSnapshot",
-    "storybook": "start-storybook -p 9000 --ci",
+    "storybook": "start-storybook -p 9000",
```
[the remark this answers is not in the document]
Comment (mimarz) [reply packages/eds-core-react/package.json:36]: Alright. Rolling back 👍

File: packages/eds-core-react/src/components/Icon/Icon.docs.mdx (hunk 1 above)
Comment (mimarz) [reply packages/eds-core-react/src/components/Icon/Icon.docs.mdx:47]: `aria-hidden="true"` is on applied by default and removed when `title` is applied. I'll see if I can reword it somehow.

**Comment by mimarz** [comment]
I still have a problem with type declarations and .mdx files. Any ideas? I've added the `mdx.d.ts` without avail.

image

**Review by vnys — CHANGES_REQUESTED** [review]
Having dedicated components for stories is a huge improvement 👍🏻 
Could we improve import statements with a path alias? 
Also PascalCasing for `Stack.tsx` and `Story.tsx`.

**Comment by mimarz** [comment]
> Having dedicated components for stories is a huge improvement 👍🏻 Could we improve import statements with a path alias? Also PascalCasing for `Stack.tsx` and `Story.tsx`.

We removed all path aliases not long ago as it was causing a lot of problems with either eslint or the generated type not being properly updated during type generation. Maybe it could work better now with the new build setup but it should be a separate issue then :) #1242

**Comment by vnys** [comment]
> We removed all path aliases not long ago as

Right, I thought something like that must have happened as I couldn’t find any other aliases.

**Review by vnys — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**